docs/index.html

```diff
@@ -20,396 +20,490 @@
+                    const dirHandle = await window.showDirectoryPicker();
+                    setImageFolderHandle(dirHandle);
                 } catch (error) {
-                    console.error("Error running Python code:", error);
+                    console.error("Error accessing folder:", error);
                 }
-            }
-        };
+            };
+
+            useEffect(() => {
+                if (imageFolderHandle) {
+                    window.saveToDB('imageFolderDirHandle', imageFolderHandle);
+                    updateFileList(imageFolderHandle, setImageList, setIsLoadingImages); // Update image list
+                    console.log("Mounted folder:", imageFolderHandle.name);
+                }
+            }, [imageFolderHandle]);
+
+            // const checkAnnotationURL = async () => {
+            //     const intervalId = setInterval(() => {
+            //         const url = localStorage.getItem('annotationURL');
+            //         if (url) {
+            //             setAnnotationURL(url);
+            //             clearInterval(intervalId); // Stop checking once the URL is available
+            //         }
+            //     }, 1000);
+            // };
 
-        const shareAnnotationURL = async () => {
-            if (!annotationURL) return;
+            const deployAnnotationSession = async () => {
+                if (imageFolderHandle) {
+                    try {
+                        const annotationsFolder = await imageFolderHandle.getDirectoryHandle('annotations', { create: true });
+                        setAnnotationsFolderHandle(annotationsFolder);
+                        await updateFileList(annotationsFolder, setAnnotationsList, setIsLoadingAnnotations);
+                        console.log("Created annotations folder:", annotationsFolder.name);
+                    } catch (error) {
+                        console.error("Error creating annotations folder:", error);
+                    }
+                    // Run code to start annotation session
+                    const code = await fetch(`./${REGISTER_SERVICE_FILE}`).then(response => response.text());
+                    try {
+                        setIsRunning(true);
+                        await workerPromise;
+                        await runCode(code);
+                        const server_url = serverURL;
+                        const token = localStorage.getItem('token');
+                        if (!token) {
+                            alert("Please login first");
+                            return;
+                        }
+                        const outputs = await runCode(`await register_service("${server_url}", "${token}")`);
+                        // find an output which contains type=service
+                        const serviceOutput = outputs.find(output => output.type === 'service');
```

Comment:
The worker has a mechanism to detect service, so we use the type 'service' to capture directly the registered service instead of parsing the stdout (this is a prefered approach), once you get the service url, you can then create an annotation url in js.